index.js

```diff
@@ -0,0 +1,238 @@
+'use strict';
+
+const TelegramBot = require('node-telegram-bot-api');
+const Matrix = require('./matrixClass.js');
+const { token } = require('./token.json');
+const answ = require('./answers.json');
+
+if (token.split(':').length !== 2 || token.split(':')[1].length !== 35) {
+  throw new Error('The variable token seems to be malformatted.');
+}
+
+const bot = new TelegramBot(token, { polling: true });
+const matrices = {};
+
+const isComand = msg => msg.text[0] === '/';
+const IsThereMatr = (chatId, name) => {
+  if (matrices[chatId] && matrices[chatId][name]) {
+    return true;
+  }
+  return false;
+};
+
+const matrixToText = matr => {
+  let res = '';
+  for (const row of matr) {
+    res += row.join('  ') + '\n';
+  }
+  return res;
+};
+
+const systemToText = matr => {
+  const rowNum = matr.length;
+  const columnNum = matr[0].length;
+  let text = 'Is your system:\n ';
+  for (let i = 0; i < rowNum; i++) {
+    for (let j = 0; j < columnNum - 2; j++) {
+      text += `(${matr[i][j]})*X${j + 1} + `;
+    }
+    text += `(${matr[i][columnNum - 2]})*X${columnNum - 1} `;
+    text += `= ${matr[i][columnNum - 1]}\n  `;
+  }
+  return text;
+};
+
+const printUsersMatr = msg => {
+  const chatId = msg.chat.id;
+  let text = 'Your matrices:\n';
+  if (!matrices[chatId]) {
+    bot.sendMessage(chatId, 'Your list of matrices is empty');
+  } else {
+    for (const name in matrices[chatId]) {
+      text += `${name}:\n`;
+      const matr = matrices[chatId][name];
+      text += matrixToText(matr.matrix);
+    }
+    bot.sendMessage(chatId, text);
+  }
+};
+
+const determinant = (msg, match) => {
+  const chatId = msg.chat.id;
+  const name = match[1];
+  if (!IsThereMatr(chatId, name)) {
+    bot.sendMessage(chatId, answ.matrNotExist);
+  } else {
+    const matr = matrices[chatId][name];
+    const det = matr.getDet();
+    bot.sendMessage(chatId, `${det}`);
+  }
+};
+
+const printMatr = (msg, match) => {
+  const chatId = msg.chat.id;
+  const name = match[1];
+  if (!IsThereMatr(chatId, name)) {
+    bot.sendMessage(chatId, answ.matrNotExist);
+  } else {
+    const matr = matrices[chatId][name];
+    const text = matrixToText(matr.matrix);
+    bot.sendMessage(chatId, text);
+  }
+};
+
+const transposeMatr = (msg, match) => {
+  const chatId = msg.chat.id;
+  const name = match[1];
+  if (!IsThereMatr(chatId, name)) {
+    bot.sendMessage(chatId, answ.matrNotExist);
+  } else {
+    const matr = matrices[chatId][name];
+    const transposedMatr = matr.transpose();
+    const text = matrixToText(transposedMatr.matrix);
+    bot.sendMessage(chatId, text);
+  }
+};
+
+const invertMatr = (msg, match) => {
+  const chatId = msg.chat.id;
+  const name = match[1];
```

Comment:
Each of the functions performs the same 'argument-extraction', it would be better to at least encapsulate it in some function. Or better have a wrapper function that would separate and extract arguments from a user query and call these functions with just arguments.
i.e.
```js
const commandWrapper = (handler, msg, match) => {
  const chatId = msg.chat.id;
  const result = handler(chatId, ...match.slice(1));
  bot.sendMessage(chatId, result);
};
```